pandas/tests/indexes/test_numeric.py

```diff
@@ -919,6 +919,10 @@ def test_constructor(self):
         res = Index([1, 2**63])
         tm.assert_index_equal(res, idx)
 
+        idx = Index([-1, 2**63], dtype=object)
```

Comment:
@jreback : Raising on `NaN` causes test breakages.  Two of them are in `test_where` for `Index` because the construction of the test involves constructing an `Index`with integer dtype but we are passing in a `NaN` as one of the elements.  The other one is a little more concerning in `test_analytics` for `Series`:
~~~python
>>> mi = MultiIndex.from_arrays([list('aabbcc'), [1, 2, 2, nan, 1, 2]])
>>> ts = Series(np.arange(len(mi)), index=mi)
>>> ts.count(level=1)
...
ValueError: cannot convert float NaN to integer
~~~